### ISSUE TITLE
difference in compiled output for star.tbf1 program

### ISSUE THREAD

**Issue by rain-1** [kragen/stoneknifeforth#5]
Hello!

I created some terminology:
* t1: running a tbf1 script in the python prototype
* t2: compiling a tbf1 script with the tinyboot1.tbf1 script running on top of the python prototype. then running that binary.
* t3: compiling a tbf1 script with the tinyboot1 binary. then running that binary.

I noticed that star.tbf1 works in t1 and t2 but miscompiled in t3.

https://github.com/rain-1/stoneknifeforth/commit/9b3d93f8b528cc4a125b799bb382a166431e19f6

Basically it seems to be missing a chunk at the end. I'm not sure why.

**Comment by rain-1** [comment]
segfault seems to only happen when the file does not end with a newline.

**Comment by tekknolagi** [comment]
Does that mean that the compiler written in tbf1 is incorrect?
